samza-core/src/main/java/org/apache/samza/container/grouper/stream/AllSspToSingleTaskGrouperFactory.java

```diff
@@ -19,51 +19,87 @@
- * partitions being consumed by a task. Hence, it is assumed that there is only 1 task that processes all messages,
- * irrespective of which partition it belongs to.
- * This also implies that container and tasks are synonymous when this grouper is used. Taskname(s) has to be globally
- * unique within a given job.
+ * AllSspToSingleTaskGrouper creates TaskInstances equal to the number of containers and assigns all partitions to be
+ * consumed by each TaskInstance. This is useful, in case of using load-balanced consumers like the high-level Kafka
+ * consumer and Kinesis consumer, where Samza doesn't control the partitions being consumed by the task.
  *
- * Note: This grouper does not take in broadcast streams yet.
+ * Note:
+ * 1. This grouper does not take in broadcast streams yet.
+ * 2. This grouper is supported only with Yarn, Zk-standalone and Passthrough JobCoordinators.
  */
+
 class AllSspToSingleTaskGrouper implements SystemStreamPartitionGrouper {
-  private final int containerId;
+  private final Set<String> processorNames;
 
-  public AllSspToSingleTaskGrouper(int containerId) {
-    this.containerId = containerId;
+  AllSspToSingleTaskGrouper(Set<String> processorNames) {
+    this.processorNames = processorNames;
   }
 
   @Override
   public Map<TaskName, Set<SystemStreamPartition>> group(final Set<SystemStreamPartition> ssps) {
+    Map<TaskName, Set<SystemStreamPartition>> groupedMap = new HashMap<>();
+
     if (ssps == null) {
       throw new SamzaException("ssp set cannot be null!");
     }
     if (ssps.size() == 0) {
       throw new SamzaException("Cannot process stream task with no input system stream partitions");
     }
 
-    final TaskName taskName = new TaskName(String.format("Task-%s", String.valueOf(containerId)));
+    processorNames.forEach(processorName -> {
+      // Create a task name for each processor and assign all partitions to each task name.
+      final TaskName taskName = new TaskName(String.format("Task-%s", processorName));
+      groupedMap.put(taskName, ssps);
+    });
 
-    return Collections.singletonMap(taskName, ssps);
+    return groupedMap;
   }
 }
 
 public class AllSspToSingleTaskGrouperFactory implements SystemStreamPartitionGrouperFactory {
+  private static final Logger LOG = LoggerFactory.getLogger(AllSspToSingleTaskGrouperFactory.class);
+  private static final String YARN_CONTAINER_COUNT = "yarn.container.count";
+
   @Override
   public SystemStreamPartitionGrouper getSystemStreamPartitionGrouper(Config config) {
-    return new AllSspToSingleTaskGrouper(config.getInt(JobConfig.PROCESSOR_ID()));
+    if (config.containsKey(TaskConfigJava.BROADCAST_INPUT_STREAMS)) {
+      throw new ConfigException("The job configured with AllSspToSingleTaskGrouper cannot have broadcast streams.");
+    }
+
+    String jobCoordinatorFactoryClassName = config.get(JobCoordinatorConfig.JOB_COORDINATOR_FACTORY);
```

Comment:
Generally, please do not refer to the constant config names directly outside the scope of the config. Use TaskConfigJava.getBroadcastSystemStreams() and JobCoordinatorConfig.getJobCoordinatorFactoryClassName() instead.

Reply:
Sure.